src/gradio_ui.py

```diff
@@ -120,14 +120,14 @@ def pairwise(iterable):
             model_history = file_data["chat"]
             # Construct a new gradio history
             new_gradio_history = []
-            for turn in pairwise(model_history):
+            for human_turn, bot_turn in pairwise(model_history):
                 # Handle the situation where convo history may be loaded before character defs
                 if char_name == "":
                     # Grab char name from the model history
-                    char_name = turn[1].split(":")[0]
+                    char_name = bot_turn.split(":")[0]
                 # Format the user and bot utterances
-                user_turn = turn[0].replace("You :", "")
-                bot_turn = turn[1].replace(f"{char_name}:", f"**{char_name}**:")
+                user_turn = human_turn.replace("You :", "")
+                bot_turn = bot_turn.replace(f"{char_name}:", f"**{char_name}**:")
```

Comment:
I think `You :` needs to become `You: ` for this to work properly, and `**{char_name}**:` -> `**{char_name}:**` for consistency with the rest of the code, no?

Reply:
Yep. My bad, overlooked it